src/main/java/io/blamer/bot/conversation/routes/Registry.java

```diff
@@ -56,67 +61,31 @@ public class Registry implements Conversation {
   @Value("${answers.registry.description}")
   private String description;
 
-  /*
-   * @todo #19 After hub service will be created, redirect the token.
-   *   Send auth message to hub, to process the data.
-   * */
+  /**
+   * The requester.
+   */
+  private final RSocketRequester requester;
+
   @Override
   public SendMessage messageFromUpdate(final Update update) {
-    SendMessage result;
-    final String text = update.getMessage().getText();
-    final String chat = String.valueOf(
-      update.getMessage().getChatId()
-    );
+    final String chat = String.valueOf(update.getMessage().getChatId());
     try {
-      final String token = text.split(" ")[1];
-      result = resultFrom(chat, token);
-    } catch (final ArrayIndexOutOfBoundsException ex) {
-      result = resultWithError(text, chat);
+      final String token = update.getMessage().getText()
+        .replace("/registry", " ")
+        .strip();
+      return this.requester
+        .route("hub.auth")
+        .data(new RegistryRequest(token, chat))
+        .retrieveMono(RegistryResponse.class)
+        .map(auth -> new SendMessage(auth.getChat(), auth.getText()))
+        .block();
+    } catch (final RuntimeException ex) {
+      return new SendMessage(chat, "Internal error...");
```

Comment:
@l3r8yJ Also, maybe we should give a customer a chance to see full stacktrace?

WDYT

Reply:
@h1alexbel until we in dev mode, I think we can throw an exception message

Reply:
@l3r8yJ in dev mode it also will help